iceberg/iceberg-handler/src/main/java/org/apache/iceberg/mr/hive/HiveIcebergOutputCommitter.java

```diff
@@ -437,6 +440,10 @@ private void commitTable(FileIO io, ExecutorService executor, OutputTable output
       dataFiles.addAll(writeResults.dataFiles());
       deleteFiles.addAll(writeResults.deleteFiles());
       referencedDataFiles.addAll(writeResults.referencedDataFiles());
+
+      isCompaction = isCompaction || Optional.ofNullable(SessionState.get())
```

Comment:
why it's in a loop?

Reply:
Fixed. 
It's was a leftover from previous variations, previously the compaction flag was taken from job conf and there is a list of them in OutputTable.